File: 2379. Minimum Recolors to Get K Consecutive Black Blocks/optimised.py
#!/usr/bin/env python

def minimumRecolors(blocks: str, k: int) -> int:
    white = 0;

    for i in range(k):
        if blocks[i] == "W":
            white += 1

    ans = white
    
    for i in range(k, len(blocks)):
        if blocks[i - k] == "W":
            white -= 1

        if blocks[i] == "W":
            white += 1

        ans = min(ans, white)


    return ans

print(minimumRecolors("WBBWWBBWBW", 7))

File: 2379. Minimum Recolors to Get K Consecutive Black Blocks/optimised.ts
function minimumRecolors(blocks: string, k: number): number {
	let white = 0;

	for (let i = 0; i < k; i++) {
		if (blocks[i] === "W") white++;
	}

	let ans = white;

	for (let i = k; i < blocks.length; i++) {
		if (blocks[i - k] === "W") {
			white--;
		}

		if (blocks[i] === "W") {
                        white++;
                }

		ans = Math.min(ans, white);
	}

	return ans;
};

console.log(minimumRecolors("WBBWWBBWBW", 7))
